packages/devtools_app/lib/src/memory/flutter/memory_heap_tree_view.dart

```diff
@@ -263,23 +281,59 @@ class HeapTreeViewState extends State<HeapTree> with AutoDisposeMixin {
     );
   }
 
+  bool escapeHit = false;
+  TextEditingController textFieldController;
+  TextField searchField;
+
+  TextField createSearchField() {
+    // Creating new TextEditingController.
+    textFieldController = TextEditingController();
+
+    return TextField(
+      autofocus: true,
+      onChanged: (value) {
+        // TODO(terry): Extend search support beyond heat map.
+        if (controller.showHeatMap) {
+          controller.search = value;
+        }
+      },
+      controller: textFieldController,
+      decoration: const InputDecoration(
+        isDense: true,
+        border: OutlineInputBorder(),
+        labelText: 'Search',
+      ),
+    );
+  }
+
   Widget _buildSearchFilterControls() {
+    searchField = createSearchField();
+
+    final searchAndRawKeyboard = RawKeyboardListener(
+      child: searchField,
+      focusNode: FocusNode(),
+      onKey: (RawKeyEvent event) {
+        if (event is RawKeyDownEvent) {
+          if (event.logicalKey.keyId == LogicalKeyboardKey.escape.keyId) {
+            // ESCAPE key pressed clear search TextField.
+            escapeHit = true;
+            controller.search = '';
+            textFieldController.clear();
+          }
+        }
+      },
+    );
+
     return Container(
       padding: const EdgeInsets.symmetric(horizontal: 6),
       child: Row(
         mainAxisSize: MainAxisSize.min,
         mainAxisAlignment: MainAxisAlignment.end,
         children: [
-          Flexible(
-            child: OutlineButton(
-              key: searchButtonKey,
-              onPressed: _search,
-              child: Label(
-                search,
-                'Search',
-                minIncludeTextWidth: 200,
-              ),
-            ),
+          Container(
+            width: 300.0,
```

Comment:
Add a TODO to use a more adaptive layout than forcing this to 300.0

Reply:
Done.